src/PostgREST/Error.hs

```diff
@@ -161,46 +204,52 @@ instance JSON.ToJSON PgError where
 
 instance JSON.ToJSON P.UsageError where
   toJSON (P.ConnectionError e) = JSON.object [
-    "code"    .= ("" :: Text),
+    "code"    .= HasqlErrorPrefix "01",
     "message" .= ("Database connection error. Retrying the connection." :: Text),
-    "details" .= (toSL $ fromMaybe "" e :: Text)]
+    "details" .= (toSL $ fromMaybe "" e :: Text),
+    "hint"    .= JSON.Null]
   toJSON (P.SessionError e) = JSON.toJSON e -- H.Error
 
 instance JSON.ToJSON H.QueryError where
   toJSON (H.QueryError _ _ e) = JSON.toJSON e
 
 instance JSON.ToJSON H.CommandError where
-  toJSON (H.ResultError (H.ServerError c m d h)) = case toS c of
-    'P':'T':_ -> JSON.object [
-        "details" .= (fmap toS d :: Maybe Text),
-        "hint"    .= (fmap toS h :: Maybe Text)]
-
-    _         -> JSON.object [
+  toJSON (H.ResultError (H.ServerError c m d h)) = JSON.object [
         "code"    .= (toS c      :: Text),
         "message" .= (toS m      :: Text),
         "details" .= (fmap toS d :: Maybe Text),
         "hint"    .= (fmap toS h :: Maybe Text)]
 
   toJSON (H.ResultError (H.UnexpectedResult m)) = JSON.object [
-    "message" .= (m :: Text)]
+    "code"    .= HasqlErrorPrefix "02",
+    "message" .= (m :: Text),
+    "details" .= JSON.Null,
+    "hint"    .= JSON.Null]
   toJSON (H.ResultError (H.RowError i H.EndOfInput)) = JSON.object [
+    "code"    .= HasqlErrorPrefix "03",
     "message" .= ("Row error: end of input" :: Text),
     "details" .= ("Attempt to parse more columns than there are in the result" :: Text),
     "hint"    .= (("Row number " <> show i) :: Text)]
   toJSON (H.ResultError (H.RowError i H.UnexpectedNull)) = JSON.object [
+    "code"    .= HasqlErrorPrefix "04",
     "message" .= ("Row error: unexpected null" :: Text),
     "details" .= ("Attempt to parse a NULL as some value." :: Text),
     "hint"    .= (("Row number " <> show i) :: Text)]
   toJSON (H.ResultError (H.RowError i (H.ValueError d))) = JSON.object [
+    "code"    .= HasqlErrorPrefix "05",
     "message" .= ("Row error: Wrong value parser used" :: Text),
     "details" .= d,
     "hint"    .= (("Row number " <> show i) :: Text)]
   toJSON (H.ResultError (H.UnexpectedAmountOfRows i)) = JSON.object [
+    "code"    .= HasqlErrorPrefix "06",
     "message" .= ("Unexpected amount of rows" :: Text),
-    "details" .= i]
+    "details" .= i,
+    "hint"    .= JSON.Null]
   toJSON (H.ClientError d) = JSON.object [
+    "code"    .= HasqlErrorPrefix "07",
```

Comment:
As we discussed on https://github.com/PostgREST/postgrest-docs/pull/430#discussion_r696189225, this type of error is a special one that refers to the connection. I think we also need a special prefix for this one. Ideally, this would be a `PGRST000` so it's clearly identifiable.

Reply:
Regrouped the errors and this one would be the `PGRST001` error after the Connection Error from the Hasql Pool.